src/controllers/movie.js

```diff
@@ -0,0 +1,90 @@
+const { Prisma } = require("@prisma/client");
+const prisma = require("../utils/prisma");
+const movieDomain = require('../domains/movie')
+
+const getAllMovies = async (req, res) => {
+  const minRuntime = (req.query.runtimeGt ? req.query.runtimeGt : null )
+  const maxRuntime = (req.query.runtimeLt ? req.query.runtimeLt : undefined )
+  
+  const movies = await movieDomain.getAllMovies(minRuntime, maxRuntime)
+  res.status(200).json({ movies })
+}
+
+const getMovieByID = async (req, res) => {
+  const { id } = req.params
+
+  try {
+    let movie
+    if (isNaN(Number(id))) {
+      movie = await movieDomain.getMovieByTitle(id)
+    } else {
+      movie = await movieDomain.getMovieByID(id)
+    }
+    res.status(200).json({ movie })
+  } catch (e) {
```

Comment:
ok so here it looks like you've taken advantage of the try catch to throw the error

Reply:
Yep, every interaction with the database takes place inside a try catch function.